pkg/operations/kubernetesupgrade/upgrader.go

```diff
@@ -221,98 +233,126 @@ func (ku *Upgrader) upgradeAgentPools() error {
 		upgradeAgentNode.Client = ku.Client
 		upgradeAgentNode.kubeConfig = ku.kubeConfig
 
-		upgradeVMs := make(map[int]*UpgradeVM)
-
-		// Go over upgraded VMs and verify provisioning state. Delete the VM if the state is not 'Succeeded'.
-		// Such a VM will be re-created later in this function.
+		agentVMs := make(map[int]*vmInfo)
+		// Go over upgraded VMs and verify provisioning state
+		// per https://docs.microsoft.com/en-us/rest/api/compute/virtualmachines/virtualmachines-state :
+		//  - Creating: Indicates the virtual Machine is being created.
+		//  - Updating: Indicates that there is an update operation in progress on the Virtual Machine.
+		//  - Succeeded: Indicates that the operation executed on the virtual machine succeeded.
+		//  - Deleting: Indicates that the virtual machine is being deleted.
+		//  - Failed: Indicates that the update operation on the Virtual Machine failed.
+		// Delete VMs in 'bad' state. Such VMs will be re-created later in this function.
+		upgradedCount := 0
 		for _, vm := range *agentPool.UpgradedAgentVMs {
-			ku.logger.Infof("Agent VM: %s, pool name: %s on expected orchestrator version\n", *vm.Name, *agentPool.Name)
+			ku.logger.Infof("Agent VM: %s, pool name: %s on expected orchestrator version", *vm.Name, *agentPool.Name)
 			var vmProvisioningState string
 			if vm.VirtualMachineProperties != nil && vm.VirtualMachineProperties.ProvisioningState != nil {
 				vmProvisioningState = *vm.VirtualMachineProperties.ProvisioningState
 			}
+			agentIndex, _ := armhelpers.GetVMNameIndex(vm.StorageProfile.OsDisk.OsType, *vm.Name)
 
-			if vmProvisioningState != string(api.Succeeded) {
-				ku.logger.Infof("Deleting agent VM %s in provisioning state %s\n", *vm.Name, vmProvisioningState)
+			switch vmProvisioningState {
+			case "Creating":
+				fallthrough
+			case "Updating":
```

Comment:
these can be written more compactly:
 case "Creating", "Updating", "Succeeded": 
Really a matter of choice though. @jackfrancis @amanohar any opinion on which style we want?

Reply:
+1 for the succinct style. See the "Multiple Cases" here:

https://github.com/golang/go/wiki/Switch

Reply:
sure, I will change that.